diff --git a/src/ceph-volume/ceph_volume/main.py b/src/ceph-volume/ceph_volume/main.py
index f8eca65ec49..4f0ebc7ec4c 100644
--- a/src/ceph-volume/ceph_volume/main.py
+++ b/src/ceph-volume/ceph_volume/main.py
@@ -4,19 +4,10 @@ import os
 import sys
 import logging
 
+from pkg_resources import iter_entry_points as entry_points  # type: ignore
 
-# `iter_entry_points` from `pkg_resources` takes one argument whereas
-# `entry_points` from `importlib.metadata` does not.
-try:
-    from importlib.metadata import entry_points
-
-    def get_entry_points(group: str):  # type: ignore
-        return entry_points().get(group, [])  # type: ignore
-except ImportError:
-    from pkg_resources import iter_entry_points as entry_points  # type: ignore
-
-    def get_entry_points(group: str):  # type: ignore
-        return entry_points(group=group)  # type: ignore
+def get_entry_points(group: str):  # type: ignore
+    return entry_points(group=group)  # type: ignore
 
 from ceph_volume.decorators import catches
 from ceph_volume import log, devices, configuration, conf, exceptions, terminal, inventory, drive_group, activate
